### ISSUE TITLE
check_icmp: performance data output changed

### ISSUE THREAD

**Issue by cseres3** [nagios-plugins/nagios-plugins#480]
Hi,
at some point the order performance data of check_icmp has changed, the packet loss value is now the last one when it used to be second. This causes pnp4nagios to show rtmax values instead of packet loss, for instance.
[RedHat bug with more detailed information](https://bugzilla.redhat.com/show_bug.cgi?id=1707814)
Any plans to change the order back how it used to be for compatibility or is there some reason for this change?
With best regards,
Cristian Seres

**Comment by sawolf** [comment]
Hi @cseres3, thanks for reporting this.

In my own opinion, I would consider this an issue with pnp4nagios rather than with check_icmp, because the order of the performance data shouldn't matter. As far as changing the data back, I think that anyone who started using the plugin with pnp4nagios after 2017 will simply have the same issue that you're having, just in reverse. 

What I'll recommend instead is that you use a wrapping script to modify the output of check_icmp, i.e. you'd switch to root and do something like
```
mv check_icmp check_icmp_binary
touch check_icmp
chown nagios:nagios 
chmod 755 check_icmp
```
and then change the contents of check_icmp to be
```
#!/bin/bash
abs_path=$(dirname $0)

$abs_path/check_icmp_binary $@ | sed 's/\([^|]*\)|rta=\(.*\) rtmax=\([^ ]*\) rtmin=\(.*\) pl=\(.*\)/\1 | rta=\2 pl=\5 rtmax=\3 rtmin=\4/'
```

This gives output like
```
[root@localhost libexec]# ./check_icmp -H 127.0.0.1
OK - 127.0.0.1 rta 0.011ms lost 0% | rta=0.011ms;200.000;500.000;0; pl=0%;40;80;0;100  rtmax=0.035ms;;;; rtmin=0.004ms;;;;
```
which seems to be what you're after.

**Comment by cseres3** [comment]
Thank you @sawolf, you tip helped us to solve the issue!

**Comment by sawolf** [comment]
Glad to hear it! I'll close this issue now.

**Comment by sawolf** [comment]
Just wanted to update you on this - we had a discussion in the office and we are going to change it back for 2.3.1. I assumed the change was released in 2.2.1 (that is, the release you can download from this repo) but in reality the change was only present in downstream patches.
